### PR TITLE
[skip ci] Skip some build checks in PRs

### PR DESCRIPTION
**Pull request by scb261**
Probably needs some cleanup, since it's my first time doing this. You can test changes by PRing to the `pr_individual` branch in my repo.

For individual sources it builds only languages that were changed. For multisrc sources it build everything when at least one file was changed in `multisrc` directory. If something was changed outside of `src` and `multisrc` - it builds everything.

Examples:
- https://github.com/scb261/tachiyomi-extensions/pull/8
- https://github.com/scb261/tachiyomi-extensions/pull/9
- https://github.com/scb261/tachiyomi-extensions/pull/10

### CODE REVIEW

**Comment by scb261** [comment]
Converting to draft for now, since it doesn't consider if something like gradle file was changed outside of src directory

**Comment by scb261** [comment]
Should be fine now, example - https://github.com/scb261/tachiyomi-extensions/pull/6

File: .github/workflows/build_pull_request.yml [hunk 1]
```diff
@@ -28,14 +28,22 @@ jobs:
         id: parsed
         run: |
           all_langs=()
+          is_root_cganged=0
```
Comment (FourTOne5) [inline .github/workflows/build_pull_request.yml:31]: `cganged` xD

File: .github/workflows/build_pull_request.yml [hunk 2]
```diff
@@ -28,14 +28,22 @@ jobs:
...
           for changed_file in ${{ steps.files.outputs.all }}; do
             if [[ ${changed_file} == src/* ]]; then
-               lang=$(echo ${changed_file} | cut -d/ -f2)
-               all_langs+=($lang)
+              lang=$(echo ${changed_file} | cut -d/ -f2)
+              all_langs+=($lang)
+            elif [[ ${changed_file} != multisrc/* ]]; then
+              is_root_cganged=1
```
Comment (FourTOne5) [inline .github/workflows/build_pull_request.yml:37]: `cganged`?

File: .github/workflows/build_pull_request.yml [hunk 3]
```diff
@@ -28,14 +28,22 @@ jobs:
...
+              break
             fi
           done
-          unique_langs=($(echo "${all_langs[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' '))
-          json_langs=$(jq --compact-output --null-input '$ARGS.positional' --args "${unique_langs[@]}")
+          if [ $is_root_cganged -eq 1 ] ; then
```
Comment (FourTOne5) [inline .github/workflows/build_pull_request.yml:41]: (⊙_◎)

**Review by FourTOne5 — CHANGES_REQUESTED** [review]

File: .github/workflows/build_pull_request.yml (hunk 1 above)
Comment (scb261) [reply .github/workflows/build_pull_request.yml]: thanks, fixed

**Comment by arkon** [comment]
Considering we don't do it by languages anymore, it may be worth just doing it so that it's:

- If only `src/` files changed: build individual
- If only `multisrc/` files changed: build multisrc
- If anything outside of those changed, build everything

So more or less what you have, but without the individual language bit?

**Comment by scb261** [comment]
Superseded by #12608

**Pull request closed without merging**